pkg/scheduler/core/scheduling_queue.go

```diff
@@ -114,6 +119,10 @@ func (f *FIFO) Pop() (*v1.Pod, error) {
 		result = obj
 		return nil
 	})
+	// if FIFO is closed, Pop() gets a nil.
+	if result == nil {
```

Comment:
It looks like Pop returns `FIFOClosedError` and we should check for that. In fact, the function we pass in doesn't have to do anything. We could just write,
```golang
result, err := f.FIFO.Pop(func(interface{}) error { return nil })

Reply:
+1

Reply:
+1.

Given that we just pass into a no-op func, the err we get back can _only_ be `FIFOClosedError`, so don't need to check err type. Additionally, would prefer to print out err like "scheduling queue is closed" over err get from client-go which is "DeltaFIFO: manipulating with closed queue".

Reply:
A) that's because we pass in a no-op function
B) this is a good point
C) agree.